src/components/UserSelector.tsx

```diff
@@ -9,8 +27,14 @@ export const UserSelector: React.FC = () => {
           className="button"
           aria-haspopup="true"
           aria-controls="dropdown-menu"
+          onClick={() => setShowUserList(!showUserList)}
+          onBlur={() => setTimeout(() => setShowUserList(!showUserList), 200)}
```

Comment:
The `onBlur` event handler uses a `setTimeout` to toggle the user list visibility. This approach might cause unexpected behavior if the user interacts with the dropdown quickly. Consider using a more reliable method to handle dropdown visibility, such as managing focus events more precisely.